src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,77 @@
+import { KeyboardEvent, ReactElement, useRef } from "react"
+import { TabList } from "./components/TabList"
+import { TabProps } from "./components/Tab"
+import { TabPanel } from "./components/TabPanel"
+import { Tab } from "./types"
+
+interface TabsProps {
+  /**
+   * We constrain `children` to be element(s) with a label and children
+   * props (e.g. `Tab`). This is only structural, however.
+   **/
+  children: ReactElement<TabProps> | Array<ReactElement<TabProps>>
+  currentTab: number
+  onTab: (index: number) => void
```

Comment:
nit pick - i prefer onTabChange, since this callback is fired every time a tab changes.